controllers/v1/library/categories.js

```diff
@@ -237,21 +285,179 @@ module.exports = class LibraryCategories extends Abstract {
 	 * @returns {Array} Library categories.
 	 */
 
+	/**
+	 * @api {get} /project/v1/library/categories/list
+	 * @apiVersion 1.0.0
+	 * @apiName list
+	 * @apiGroup LibraryCategories
+	 * @apiHeader {String} X-auth-token Authenticity token
+	 * @apiUse successBody
+	 * @apiUse errorBody
+	 */
 	async list(req) {
-		return new Promise(async (resolve, reject) => {
-			try {
-				let projectCategories = await libraryCategoriesHelper.list(req)
+		try {
+			const result = await libraryCategoriesHelper.list(req)
+			return {
+				success: true,
+				message: result.message,
+				result: result.data,
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
 
-				projectCategories.result = projectCategories.data
+	// (removed) Global hierarchy endpoint: implementation removed to keep only category-specific hierarchy
```

Comment:
@Sachintechjoomla Does this comment makes any sense as hierarchy endpoint was not there previously???

Reply:
No,
Removed code